medcat/utils/ner/deid.py

```diff
@@ -62,18 +61,56 @@ def train(self, json_path: Union[str, list, None],
               *args, **kwargs) -> Tuple[Any, Any, Any]:
         return super().train(json_path, *args, train_nr=0, **kwargs)  # type: ignore
 
-    def deid_text(self, text: str, redact: bool = False) -> str:
+    def deid_text(self, text: str, redact: bool = False,config: Optional[ConfigTransformersNER] = None) -> str:
         """Deidentify text and potentially redact information.
 
+        v2: Changed to address the limit of 512 tokens. Adding chunking (break down the document into mini-documents and then run the model)
+        v3 planned: Add overlapping window instead of a straight cut
+
         Args:
             text (str): The text to deidentify.
             redact (bool): Whether to redact the information.
 
         Returns:
             str: The deidentified text.
         """
-        self.cat.get_entities
-        return deid_text(self.cat, text, redact=redact)
+
+        blocks = [[]]
+        whitespace_pattern = re.compile(r'\s')
+        maximum_tkns = min(510, config.general['maximum_tokens_model'])
```

Comment:
If we're going to have a very specific logic in here, it would need to be documented.
And if it's known not to work with other types of models, we would need to detect and (at the very least) warn users if they are using something else.

Reply:
agree - logger.warning & and comment here would be ideal if detected models from deserialized models detected to not be Roberta